### ISSUE TITLE
Release process for Immich containers?

### ISSUE THREAD

**Issue by solomon-b** [immich-app/immich#8269]
### The bug

I'm using an alternative deployment process (Nix) which relies on pinning containers by their image digest. I've noticed that non-release images are sometimes deleted from the container registry which breaks my deploy process if I am pinned to that container. 

For the main server container I can avoid the issue by sticking to images tagged with `release`, however the machine learning container does not use a `release` tag. 

Are there any machine learning containers which are guaranteed not to be deleted? If not, would it be possible to periodically create release tagged containers which are preserved?

### The OS that Immich Server is running on

Nixos

### Version of Immich Server

n/a

### Version of Immich Mobile App

n/a

### Platform with the issue

- [ ] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
n/a
```

### Your .env content

```Shell
n/a
```

### Reproduction steps

```bash
Pick a container release by image digest and wait.
```

**Comment by alextran1502** [comment]
you can pin your machine-learning container to the `release` tag as well. This is the image and the tag

`ghcr.io/immich-app/immich-machine-learning:release`

**Comment by solomon-b** [comment]
That doesn't resolve the issue. I need to reference a stable image digest but those PR builds can get deleted.

**Comment by alextran1502** [comment]
Does it need a digest?

**Comment by solomon-b** [comment]
Yes as I said in the issue. My build system requires pining to a digest.

**Comment by bo0tzz** [comment]
Does the release tag not have a stable digest? PR builds do indeed get cleaned up to avoid unnecessarily filling up the registry.

**Comment by jrasm91** [comment]
You can use whatever tag you want. The only ones that get deleted are tags specific to pull requests. If the label doesn't have the format of `pr-*` it won't get deleted.
